Sources/ETCD/ETCDClient.swift

```diff
@@ -120,5 +123,129 @@ public final class EtcdClient: @unchecked Sendable {
     public func put(_ key: String, value: String) async throws {
         try await put(key.utf8, value: value.utf8)
     }
+    
+    /// Watches a specified key in the ETCD server.
+    ///
+    /// - Parameters:
+    ///   - key: The key for which the value is watched. Parameter is of type Sequence<UInt8>.
+    ///   - operation: The operation to be run on the WatchAsyncSequence for the key.
+    public func watch<Result>(_ key: some Sequence<UInt8>, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key) }]
+        let watchAsyncSequence = WatchAsyncSequence(grpcAsyncSequence: watchClient.watch(request))
+        return try await operation(watchAsyncSequence)
+    }
+    
+    /// Watches a specified key in the ETCD server.
+    ///
+    /// - Parameters:
+    ///   - key: The key for which the value is watched. Parameter is of type String.
+    ///   - operation: The operation to be run on the WatchAsyncSequence for the key.
+    public func watch<Result>(_ key: String, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        try await self.watch(key.utf8, operation)
+    }
+}
+
+public enum EventType {
+    case put
+    case delete
+
+    init(protoType: Int) {
+        switch protoType {
+        case 0:
+            self = .put
+        case 1:
+            self = .delete
+        default:
+            fatalError("Unsupported event type")
+        }
+    }
+
+    var protoType: Int {
+        switch self {
+        case .put:
+            return 0
+        case .delete:
+            return 1
+        }
+    }
+}
+
+public struct KeyValue {
```

Comment:
Can we please move this to a separate file, give it some documentation on the type and all the fields